File: truevfs-ext/truevfs-ext-insight/src/main/scala/net/java/truevfs/ext/insight/I5tManagerDecorator.scala
/*
 * Copyright (C) 2005-2013 Schlichtherle IT Services.
 * All rights reserved. Use is subject to license terms.
 */
package net.java.truevfs.ext.insight

import net.java.truevfs.kernel.spec._
import net.java.truevfs.kernel.spec.spi._

/**
  * @author Christian Schlichtherle
  */
@deprecated("This class is reserved for exclusive use by the [[net.java.truevfs.kernel.spec.sl.FsManagerLocator.SINGLETON]]!", "1")
final class I5tManagerDecorator
extends FsManagerDecorator with Immutable {

  override def apply(manager: FsManager): FsManager =
    syncOperationsMediator.instrument(manager)

  /** @return -200 */
  override def getPriority = -200
}

File: truevfs-ext/truevfs-ext-insight/src/main/java/net/java/truevfs/ext/insight/I5tManagerDecorator.java
/*
 * Copyright (C) 2005-2020 Schlichtherle IT Services.
 * All rights reserved. Use is subject to license terms.
 */
package net.java.truevfs.ext.insight;

import net.java.truevfs.kernel.spec.FsManager;
import net.java.truevfs.kernel.spec.spi.FsManagerDecorator;

import static net.java.truevfs.ext.insight.I5tMediators.syncOperationsMediator;

/**
 * @author Christian Schlichtherle
 * @deprecated This class is reserved for exclusive use by the {@link net.java.truevfs.kernel.spec.sl.FsManagerLocator}
 * singleton!
 */
@Deprecated(since = "1")
public final class I5tManagerDecorator extends FsManagerDecorator {

    @Override
    public FsManager apply(FsManager manager) {
        return syncOperationsMediator.instrument(manager);
    }

    /**
     * Returns {@code -200}.
     */
    @Override
    public int getPriority() {
        return -200;
    }
}
